frontend/src/app/modules/games/games-saga.js

```diff
@@ -11,11 +13,21 @@ export function* getGamesSaga() {
   }
 }
 
-export function* selectGameSaga(action) {
-  yield put(GamesActions.Creators.selectGame(action.games[0]))
+export function* selectGameSaga(game) {
+  yield put(GamesActions.Creators.selectGame(game))
+}
+
+export function* selectGameOnGamesLoadedSaga(action) {
+  yield call(selectGameSaga, action.games[0])
+}
+
+export function* refreshSelectionSaga() {
+  const selectedGame = yield select(getSelectedGame)
+  yield call(selectGameSaga, selectedGame)
 }
 
 export function* gamesSaga() {
   yield fork(getGamesSaga)
-  yield takeEvery(GamesActions.Types.GAMES_LOADED, selectGameSaga)
+  yield takeEvery(GamesActions.Types.GAMES_LOADED, selectGameOnGamesLoadedSaga)
```

Comment:
this is a matter of taste, feel free to keep it like it is. I'd probably consider different names to these sagas, namely onGamesLoadedSaga and onMatchAddedSaga. But this is fine as well.

Reply:
can't we put another saga on the same Action?

Reply:
Theoretically we can; practically I found it can get tricky to take the same action from multiple places as there might be concurrency issues and also both sagas could to some overlapping operation (both are a result on bad architecture of course). However, like it said, this is fine as well.

Reply:
ok, let's leave as it is but I see your point